src/external/SMIOL/smiol.c

```diff
@@ -336,10 +336,12 @@ int SMIOL_open_file(struct SMIOL_context *context, const char *filename,
 				MPI_Comm_free(&io_group_comm);
 				return SMIOL_INVALID_FORMAT;
 			}
-
+            MPI_Info info;
```

Comment:
I'd suggest following the convention in SMIOL of declaring variables at the beginning of code blocks (beginning on line 323 in this case).